input/include/NovelRT/Input/IInputDevice.hpp

```diff
@@ -29,6 +32,9 @@ namespace NovelRT::Input
         [[nodiscard]] virtual NovelKey& GetAvailableKey(const std::string& keyRequested) = 0;
         [[nodiscard]] virtual NovelRT::Maths::GeoVector2F GetMousePosition() = 0;
         [[nodiscard]] virtual NovelRT::Utilities::Misc::Span<InputAction> GetAllMappings() = 0;
+        [[nodiscard]] virtual KeyStateFrameChangeLog GetCurrentChangeLog(const std::string& key) = 0;
+        [[nodiscard]] virtual KeyStateFrameChangeLog GetPreviousChangeLog(const std::string& key) = 0;
```

Comment:
any concerns about mutability here? I'm fine to leave it as is given its pure virtual but wanted your thoughts

Reply:
Nay, no concerns here